### PR TITLE
Updating training script in basic_interactive notebook

### PR DESCRIPTION
**Pull request by Fiona-Waters**
# Issue link
[RHOAIENG-6475](https://issues.redhat.com/browse/RHOAIENG-6475)

# What changes have been made
When running GPU utilising workloads using the basic_interactive demo notebook I encountered a lot of errors while running the training script provided in the notebook. As it was added more than a year ago and may be out of date I have updated the notebook to use ray train using these examples: [Text classification](https://huggingface.co/docs/transformers/tasks/sequence_classification ), [Hugging Face Transformers + Ray Train ](https://docs.ray.io/en/latest/train/getting-started-transformers.html)

I have also updated the hf_interactive notebook which was using the same training script.

# Verification steps
 - Install RHOAI with the CodeFlare components set to `Managed`
 - Ensure that your cluster has GPU nodes as required.
 - Within a data science project workbench clone this branch and install codeflare sdk
```
git clone https://github.com/Fiona-Waters/codeflare-sdk.git -b udpate-notebook-script
cd codeflare-sdk
pip install poetry
poetry install
```
 - Run the basic_interactive notebook and ensure that it works as expected.
 - Note that aws credentials are required and you will need to create and use an s3 bucket while running the training script. See this [PR](https://github.com/project-codeflare/codeflare-sdk/pull/563/files) for instructions.

## Checks
- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [X] Manual tests
   - [ ] Testing is not required for this change

### CODE REVIEW

File: demo-notebooks/additional-demos/hf_interactive.ipynb [hunk 1]
```diff
@@ -100,7 +98,7 @@
     "                                       max_cpus=8, \n",
     "                                       min_memory=16, \n",
     "                                       max_memory=16, \n",
-    "                                       image=\"quay.io/project-codeflare/ray:latest-py39-cu118\",\n",
+    "                                       image=\"quay.io/project-codeflare/ray:2.20.0-py39-cu118\",\n",
```
Comment (Bobbins228) [inline demo-notebooks/additional-demos/hf_interactive.ipynb:101]:
```suggestion
    "                                       image=\"quay.io/project-codeflare/ray:latest-py39-cu118\",\n",
```
We can put this back for now. I plan to update all of the demo notebooks with new images once the ray version PR #530 is in

File: demo-notebooks/guided-demos/2_basic_interactive.ipynb [hunk 2]
```diff
@@ -69,7 +69,7 @@
     "    max_cpus=2,\n",
     "    min_memory=8,\n",
     "    max_memory=8,\n",
-    "    image=\"quay.io/project-codeflare/ray:latest-py39-cu118\",\n",
+    "    image=\"quay.io/project-codeflare/ray:2.20.0-py39-cu118\",\n",
```
Comment (Bobbins228) [inline demo-notebooks/guided-demos/2_basic_interactive.ipynb:72]:
```suggestion
    "    image=\"quay.io/project-codeflare/ray:latest-py39-cu118\",\n",
```

File: demo-notebooks/guided-demos/notebook-ex-outputs/2_basic_interactive.ipynb [hunk 3]
```diff
@@ -77,7 +77,7 @@
     "    max_cpus=2,\n",
     "    min_memory=8,\n",
     "    max_memory=8,\n",
-    "    image=\"quay.io/project-codeflare/ray:latest-py39-cu118\",\n",
+    "    image=\"quay.io/project-codeflare/ray:2.20.0-py39-cu118\",\n",
```
Comment (Bobbins228) [inline demo-notebooks/guided-demos/notebook-ex-outputs/2_basic_interactive.ipynb:80]:
```suggestion
    "    image=\"quay.io/project-codeflare/ray:latest-py39-cu118\",\n",
```

**Review by Bobbins228 — CHANGES_REQUESTED** [review]
I had to test with the `latest-py39-cu118` image. It worked like a charm. 
Happy to lgtm once the images are set back. 
Awesome work Fiona :rocket:

File: demo-notebooks/additional-demos/hf_interactive.ipynb (hunk 1 above)
Comment (Fiona-Waters) [reply demo-notebooks/additional-demos/hf_interactive.ipynb]: Perfect, I've reverted it in all places. Thanks @Bobbins228

**Review by Bobbins228 — COMMENTED** [review]
/lgtm

File: demo-notebooks/additional-demos/hf_interactive.ipynb [hunk 4]
```diff
@@ -317,14 +315,13 @@
...
     "import ray\n",
-    "from ray.air.config import ScalingConfig\n",
     "\n",
     "# reset the ray context in case there's already one. \n",
     "ray.shutdown()\n",
     "# establish connection to ray cluster\n",
     "\n",
     "#install additional libraries that will be required for this training\n",
-    "runtime_env = {\"pip\": [\"transformers\", \"datasets\", \"evaluate\", \"pyarrow<7.0.0\", \"accelerate\"]}\n",
+    "runtime_env = {\"pip\": [\"transformers\", \"datasets==2.17.0\", \"pyarrow>7.0.0\", \"accelerate\", \"scikit-learn\"]}\n",
```
Comment (astefanutti) [inline demo-notebooks/additional-demos/hf_interactive.ipynb:324]: Can these be all pinned? Is `pyarrow` a direct dependency?

Comment (Fiona-Waters) [reply demo-notebooks/additional-demos/hf_interactive.ipynb:324]: I can pin them if that's better. 
A direct dependency of the sdk? No, just needed to run this training script. Apologies if I misunderstand what you are asking.

**Comment by Bobbins228** [comment]
This PR just needs a rebase and it should pass the e2e test

File: demo-notebooks/additional-demos/hf_interactive.ipynb (hunk 4 above)
Comment (Fiona-Waters) [reply demo-notebooks/additional-demos/hf_interactive.ipynb]: Ok looked at it again, have removed pyarrow as it is already installed with sdk dependencies and have pinned the others. Thanks @astefanutti :)

File: demo-notebooks/additional-demos/hf_interactive.ipynb [hunk 5]
```diff
@@ -317,14 +315,13 @@
...
     "import ray\n",
-    "from ray.air.config import ScalingConfig\n",
     "\n",
     "# reset the ray context in case there's already one. \n",
     "ray.shutdown()\n",
     "# establish connection to ray cluster\n",
     "\n",
     "#install additional libraries that will be required for this training\n",
-    "runtime_env = {\"pip\": [\"transformers\", \"datasets\", \"evaluate\", \"pyarrow<7.0.0\", \"accelerate\"]}\n",
+    "runtime_env = {\"pip\": [\"transformers==4.41.2\", \"datasets==2.17.0\", \"accelerate==0.31.0\", \"scikit-learn==1.5.0\"]}\n",
```
Comment (astefanutti) [inline demo-notebooks/additional-demos/hf_interactive.ipynb:324]: Last nit: is `scikit-learn` actually used?

Comment (Fiona-Waters) [reply demo-notebooks/additional-demos/hf_interactive.ipynb:324]: Yes, without it I get an error that it is required. I've cleaned up my notebook but can run it again without it if you'd like to see the error.

Comment (astefanutti) [reply demo-notebooks/additional-demos/hf_interactive.ipynb:324]: Out of curiosity, what error was that? Maybe `numpy`?

Comment (Fiona-Waters) [reply demo-notebooks/additional-demos/hf_interactive.ipynb:324]: Its related to "accuracy" in the compute_metrics function. The error is:
`ImportError: To be able to use accuracy, you need to install the following dependency: sklearn.
Please install it using 'pip install scikit-learn' for instance.`

Comment (astefanutti) [reply demo-notebooks/additional-demos/hf_interactive.ipynb:324]: Ah right, thanks!

**Review by Bobbins228 — APPROVED** [review]

**Pull request merged**